qiskit/providers/__init__.py

```diff
@@ -110,6 +109,381 @@
+
+    from qiskit.providers import BackendV1 as Backend
+    from qiskit.providers.models import BackendConfiguration
+    from qiskit.providers import Options
+
+
+    class MVbackend(Backend):
+
+        configuration = {
+            'backend_name': 'my_backend',
+            'backend_version': '1.0.0',
+            'n_qubits': 5,
+            'basis_gates': ['p', 'sx', 'u', 'cx', 'id'],
+            'coupling_map': [[0, 1], [1, 2], [2, 3], [3. 4]],
+            'simulator': False,
+            'local': False,
+            'conditional': True,
+            'open_pulse': False,
+            'memory': True,
+            'max_shots': 8196,
+            'gates': []
+        }
+
+        def __init(self, provider):
+            super().__init__(
+                configuration=BackendConfiguration.from_dict(self.configuration),
+                provider=provider)
+
+        @classmethod
+        def _default_options(cls):
+            return Options(shots=1024, memory=False)
+
+        def run(circuits, **kwargs):
+            # serialize circuits submit to backend and create a job
+            for kwarg in kwargs:
+                if not hasattr(kwarg, self.options):
+                    warnings.warn(
+                        "Option %s is not used by this backend" % kwarg,
+                        UserWarning, stacklevel=2)
+            options = {
+                'shots': kwargs.get('shots', self.options.shots)
+                'memory': kwargs.get('memory', self.options.shots),
+            }
+            job_json = convert_to_wire_format(circuit, options)
+            job_handle = submit_to_backend(job_jsonb)
+            return MyJob(self. job_handle, job_json, circuit)
+
+
+Transpiler Interface
+--------------------
+
+The key piece of the backend object is how it describes itself to the compiler.
+This is composed of three classes, the
+:class:`~qiskit.providers.models.BackendConfiguration` which describes the
+immutable properties of the backend (things like the number of qubits, coupling
+map, etc), the :class:`~qiskit.providers.models.BackendProperties` which
+described measured properties of the device (such as current error rates), and
+the :class:`~qiskit.providers.models.PulseDefaults` which defines the
+default pulse behavior for the backend (such as the pulse sequences for gates).
+Of these the only required piece for a backend is the
+:class:`~qiskit.providers.models.BackendConfiguration`, the other information
+is only used if the backend either returns the necessary information.
+
+It's worth noting that this interface will likely evolve over time as the
+format, details, API, and information of the data the compiler needs from a
+backend will change as the transpiler and backends grow and improve. This is
+a key reason why the providers interfaces are versioned to enable making these
+changes in a controlled manner. The details here will change and reflect the
+latest version of this interface.
+
+Custom Basis Gates
```

Comment:
This is very good.  However, should this not be some place else like: https://qiskit.org/documentation/stubs/qiskit.circuit.EquivalenceLibrary.html ?  Perhaps we need some here and some there, not exactly sure.

Reply:
Yeah, I wasn't sure exactly where to put it. I settled on here because this has come up a couple of times in the context of writing a provider from scratch for a backend with custom gate definitions. I figured it was good here to provide concrete instructions on how to do this and how to ensure `transpile()` and `execute()` will work for compiling to a custom basis set.